src/UserEmail.php

```diff
@@ -257,12 +257,16 @@ public static function showForUser(User $user)
         ) {
             return;
         }
-        $canedit = (
-            (
-                $user->can($users_id, UPDATE)
-                && ($user->currentUserHaveMoreRightThan($users_id)))
-            || ($users_id == Session::getLoginUserID())
-        );
+
+        $canedit = $users_id == Session::getLoginUserID();
+        if (!$canedit) {
+            if ($user->isNewID($users_id)) {
+                $canedit = $user->can($users_id, CREATE);
+            } else {
+                $canedit = $user->can($users_id, UPDATE)
+                && $user->currentUserHaveMoreRightThan($users_id);
+            }
+        }
```

Comment:
Seems OK.
Why IFs and not a single condition?
```suggestion
        $canedit = (
                $users_id == Session::getLoginUserID()
            ) || (
                $user->isNewID($users_id)
                && $user->can($users_id, CREATE)
            ) || (
                $user->can($users_id, UPDATE)
                && ($user->currentUserHaveMoreRightThan($users_id))
        );
```